profiler/options.go

```diff
@@ -208,6 +211,7 @@ func defaultConfig() (*config, error) {
 		uploadTimeout:     DefaultUploadTimeout,
 		maxGoroutinesWait: 1000, // arbitrary value, should limit STW to ~30ms
 		deltaProfiles:     internal.BoolEnv("DD_PROFILING_DELTA", true),
+		deltaMethod:       os.Getenv("DD_PROFILING_DELTA_METHOD"),
```

Comment:
In Paris, we agreed to release this behind a flag, disabled by default.  I still think we should release this in stages.

- 1.44.0 with both implementations, old algorithm as default, new algorithm enabled with an env var. This is so we can continue testing internally.  dd-go apps are not going to go to prod with a non-release dd-trace-go.  Customers that need it can opt-in.
- 1.4?.0 still include both algorithms, make fastdelta the default, enable old algorithm with an env var fallback.